WordPress/src/main/java/org/wordpress/android/ui/main/SitePickerActivity.java

```diff
@@ -180,6 +180,7 @@ private void updateMenuItemVisibility() {
 
         // no point showing search if there aren't multiple blogs
         mMenuSearch.setVisible(mSiteStore.getSitesCount() > 1);
+        invalidateOptionsMenu();
```

Comment:
Adding `invalidateOptionsMenu()` causes the menu to be refreshed multiple times.  Calling `invalidateOptionsMenu()` in `updateMenuItemVisibility()` causes `onPrepareOptionsMenu(Menu menu)` to be triggered, which calls `updateMenuItemVisibility()` and `invalidateOptionsMenu()` again.

Even though refreshing the menu multiple times fixes the bug, it seems to mask the root cause.  We should probably remove this `invalidateOptionsMenu()` call in order to address the underlying issue.

Reply:
@theck13 I looked how the Menu is used in ```MediaBrowserActivity``` and I did something similar. 

In ```disableSearchMode()``` method I removed  the call to ```updateMenuItemVisibility()``` and added ```invalidateOptionsMenu()```. This causes ```onPrepareOptionsMenu(Menu menu)``` to be triggered, which calls ``` updateMenuItemVisibility()```

I had to use ```invalidateOptionsMenu()``` since ```onPrepareOptionsMenu()``` is only called the first time and then no calls are made to this method. 

Do you think it is possible to fix this bug without using ```invalidateOptionsMenu()``` ?

Reply:
I think using `invalidateOptionsMenu()` is correct, but we shouldn't use it in `updateMenuItemVisibility()` since that will cause multiple calls as I mentioned.

I agree that replacing `updateMenuItemVisibility()` with `invalidateOptionsMenu()` in `disableSearchMode()` is right.  We could do the same thing in `enableSearchMode()`.